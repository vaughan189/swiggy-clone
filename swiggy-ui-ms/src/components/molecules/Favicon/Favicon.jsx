/**
 *
 * Favicon
 *
 */
import React from 'react';

const Favicon = () => {
  return (
    <>
      <link rel="icon" type="image/png" href="/images/favicon.ico" sizes="16x16" />
      <link rel="icon" type="image/png" href="/images/favicon.ico" sizes="32x32" />
      <link rel="icon" type="image/png" href="/images/favicon.ico" sizes="96x96" />
    </>
  );
};

export default Favicon;
