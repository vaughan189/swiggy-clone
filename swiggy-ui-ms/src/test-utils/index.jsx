import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { theme } from '../theme';

const AllTheProviders = ({ children, mocks = [] }) => {
  return (
    <MockedProvider mocks={mocks} addTypename={false}>
        {children}
    </MockedProvider>
  );
};

AllTheProviders.propTypes = {
  children: PropTypes.node.isRequired,
  mocks: PropTypes.arrayOf(PropTypes.shape({})),
};

AllTheProviders.defaultProps = {
  mocks: [],
};

const customRender = (ui, options) => {
  const { wrapperProps = {}, ...customOptions } = options || {};
  return render(ui, {
    wrapper: (props) => {
      return <AllTheProviders {...props} {...wrapperProps} />;
    },
    ...customOptions,
  });
};

// re-export everything
export * from '@testing-library/react';
export * from './withTestRouter';

// override render method
export { customRender as render };
