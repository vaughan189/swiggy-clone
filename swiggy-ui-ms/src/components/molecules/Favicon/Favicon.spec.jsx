import React from 'react';
import { render } from '../../../test-utils';

import Favicon from '.';

describe('Favicon', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Favicon />);
    expect(asFragment()).toMatchSnapshot();
  });
});
