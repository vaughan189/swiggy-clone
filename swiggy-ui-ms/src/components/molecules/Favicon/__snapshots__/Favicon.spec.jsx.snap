// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favicon should render correctly 1`] = `
<DocumentFragment>
  <link
    href="/images/favicon.ico"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />
  <link
    href="/images/favicon.ico"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />
  <link
    href="/images/favicon.ico"
    rel="icon"
    sizes="96x96"
    type="image/png"
  />
</DocumentFragment>
`;
